iceoryx2-cal/tests/reactor_tests.rs

```diff
@@ -91,6 +91,21 @@ mod reactor {
         assert_that!(sut.len(), eq 0);
     }
 
+    #[test]
+    fn attach_the_same_attachment_twice_fails<Sut: Reactor>() {
+        let sut = <<Sut as Reactor>::Builder>::new().create().unwrap();
+
+        let name = generate_name();
+        let listener = unix_datagram_socket::ListenerBuilder::new(&name)
+            .create()
+            .unwrap();
+
+        let _guard = sut.attach(&listener).unwrap();
```

Comment:
In my opinion, I prefer reading this than the `Box<dyn ...` in the test name and the `Ok(())` at the end ... It is just a personal preference.

Reply:
I have to check why I formed a habit to returning an error instead of unwrapping. It might be that the error message was not fully clear when it failed due to an unwrap but maybe that changed meanwhile